textdata_reader.py

```diff
@@ -0,0 +1,181 @@
+from mimic.onto.mimic_ontology import Impression, Findings, FilePath
+from forte.data.multi_pack import MultiPack
+from forte.processors.base import MultiPackProcessor
+from forte.data.caster import MultiPackBoxer
+from ft.onto.base_ontology import Token
+from forte.processors.writers import PackNameJsonPackWriter
+from forte.data.selector import NameMatchSelector
+
+
+class FindingsExtractor(PackProcessor):
+    r"""A wrapper of fingings extractor.
+    """
+
+    def __init__(self):
+        super().__init__()
+
+    def _process(self, input_pack: DataPack):
+        findings_ind = input_pack.text.find("FINDINGS")
+        impression_ind = input_pack.text.find("IMPRESSION")
+        if findings_ind == -1:
+            begin = 0
+            end = 0
+        else:
+            begin = findings_ind + len("FINDINGS") + 1
+            if impression_ind != -1 and findings_ind < impression_ind:
+                end = impression_ind - 1
+            else:
+                end = len(input_pack.text)
+        findings = Findings(input_pack, begin, end)
+        findings.has_content = (findings_ind != -1)
+
+
+class ImpressionExtractor(PackProcessor):
+    r"""A wrapper of impression extractor.
+    """
+
+    def __init__(self):
+        super().__init__()
+
+    def _process(self, input_pack: DataPack):
+        findings_ind = input_pack.text.find("FINDINGS")
+        impression_ind = input_pack.text.find("IMPRESSION")
+        if impression_ind == -1:
+            begin = 0
+            end = 0
+        else:
+            begin = impression_ind + len("IMPRESSION") + 1
+            if findings_ind != -1 and impression_ind < findings_ind:
+                end = findings_ind - 1
+            else:
+                end = len(input_pack.text)
+        impression = Impression(input_pack, begin, end)
+        impression.has_content = (impression_ind != -1)
+
+
+class FilePathGetter(PackProcessor):
+    r"""A wrapper to get the file path hierarchy of the current file
+    """
+    def __init__(self):
+        super().__init__()
+
+    def _process(self, input_pack: DataPack):
+        filepath = FilePath(input_pack)
+        if input_pack.pack_name is not None:
+            filepath.img_study_path = input_pack.pack_name
+        else:
+            filepath.img_study_path = "packname"
+
+
+class NonAlphaTokenRemover(MultiPackProcessor):
+    r"""A wrapper of NLTK word tokenizer.
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.in_pack_name = 'default'
+        self.out_pack_name = 'result'
+
+    def _process(self, input_pack: MultiPack):
+        token_entries = list(input_pack.get_pack(
+            self.in_pack_name).get(entry_type=Token))
+        token_texts = [token.text for token in token_entries]
+        words = [word for word in token_texts
+                 if word.isalpha() or word[:-1].isalpha() or word == '.']
+        filepath = list(input_pack.get_pack(
+            self.in_pack_name).get(entry_type=FilePath))[0]
```

Comment:
```suggestion
            self.in_pack_name).get_single(FilePath))
```
You can use the `get_single` function here.